src/gui/mzroll/eicwidget.cpp

```diff
@@ -2288,3 +2288,422 @@ void EicWidget::addMS2Events(float mzmin, float mzmax)
+                         scene.width(),
+                         scene.height(),
+                         _minX,
+                         _minY,
+                         _maxX,
+                         _maxY);
+                setLineAttributes(lineEic, eic, fadedMultiplier, zValue);
+            }
+        }
+
+    }
+
+    ////Add Peak Positions////
+    for (unsigned int i = 0; i < group->peaks.size(); i++) {
+        Peak& peak = group->peaks[i];
+
+        if (peak.getSample() != NULL && peak.getSample()->isSelected == false)
+            continue;
+
+        QColor color = Qt::black;
+        if (peak.getSample() != NULL) {
+            mzSample* s = peak.getSample();
+            color = QColor::fromRgbF(s->color[0], s->color[1], s->color[2],
+                                     s->color[3]);
+        }
+
+        QPen pen(color, 1, Qt::SolidLine, Qt::RoundCap, Qt::RoundJoin);
+        QBrush brush(color);
+        brush.setStyle(Qt::NoBrush);
+
+        EicPoint* p = new EicPoint(((peak.rt - _minX) / (_maxX - _minX) * scene.width()),
+                                   scene.height() - ((peak.peakIntensity - _minY) / (_maxY - _minY) * scene.height()),
+                                   &peak,
+                                   getMainWindow());
+        p->setColor(color);
+        p->forceFillColor(true);
+        p->setBrush(brush);
+        p->setPen(pen);
+        p->setPeakGroup(group);
+        scene.addItem(p);
+    }
+
+    ////Add Selection Line////
+    auto selectionLine = new QGraphicsLineItem(nullptr);
+    scene.addItem(selectionLine);
+    if (rtMin < _minX)
+        rtMin = _minX;
+    if (rtMax > _maxX)
+        rtMax = _maxX;
+
+    QPen pen(Qt::red, 3, Qt::SolidLine, Qt::FlatCap, Qt::RoundJoin);
+    selectionLine->setPen(pen);
+    selectionLine->setZValue(1000);
+    selectionLine->setLine(((rtMin - _minX) / (_maxX - _minX) * scene.width()),
+                           scene.height(),
+                           (rtMax - _minX) / (_maxX - _minX) * scene.width(), scene.height());
+    selectionLine->update();
+
+    ////Add Focus Line////
+    float rt;
+    if (group && group->getCompound() != NULL && group->getCompound()->expectedRt() > 0)
+        rt = group->getCompound()->expectedRt();
+    auto focusLine = new QGraphicsLineItem(0);
+    scene.addItem(focusLine);
+    QPen penFocus(Qt::red, 2, Qt::DashLine, Qt::RoundCap, Qt::RoundJoin);
+    focusLine->setPen(penFocus);
+    focusLine->setLine(((rt - _minX) / (_maxX - _minX) * scene.width()),
+                       0,
+                       ((rt - _minX) / (_maxX - _minX) * scene.width()),
+                       scene.height());
+
+    ////Barplot////
+    auto barplot = new BarPlot(NULL, 0);
+    scene.addItem(barplot);
+
+    barplot->setMainWindow(getMainWindow());
+    barplot->setPeakGroup(group);
+
+    int bwidth = barplot->boundingRect().width();
+    int bheight = barplot->boundingRect().height();
+    int xpos = scene.width() * 0.95 - 130;
```

Comment:
I think this hardcoded `- 130` offset does not work for all devices. For instance, I am getting this kind of an image:
<img width="1356" alt="Screenshot 2020-05-06 at 3 10 48 PM" src="https://user-images.githubusercontent.com/10397162/81165002-db076d80-8fae-11ea-9960-f2f87faf8afe.png">
The bars for the highest intensity peaks got clipped out.

Use a dynamic offset, maybe using the barplot item's overall width itself.